### PR TITLE
Alerting: New alert state list view

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

This PR includes several changes;

**Building blocks**

Abstract building blocks (such as `ListItem`,  `ListGroup` and `ListSection`) to prepare for a larger list view update coming later this year.

**Refactor**

A refactor to use these for the existing components (`Namespace`, `AlertRuleListItem`, etc.) that are included in the v2 list view (currently behind a feature flag and a WIP)

**New state view**

Even though we plan to deprecate this view we're using it as a proving ground for these new components.

image

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx [hunk 1]
```diff
@@ -398,23 +396,20 @@ function usePageNav(rule: CombinedRule) {
...
 
-const calculateTotalInstances = (stats: CombinedRule['instanceTotals']) => {
+export const calculateTotalInstances = (stats: CombinedRule['instanceTotals']) => {
   return chain(stats)
-    .pick([AlertInstanceTotalState.Alerting, AlertInstanceTotalState.Pending, AlertInstanceTotalState.Normal])
+    .pick([
+      AlertInstanceTotalState.Alerting,
+      AlertInstanceTotalState.Pending,
+      AlertInstanceTotalState.Normal,
+      AlertInstanceTotalState.NoData,
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx:405]: why we add NoData and Error ? Is this is intentional?

File: public/app/features/alerting/unified/components/rules/RuleListStateView.tsx [hunk 2]
```diff
@@ -32,34 +40,97 @@ export const RuleListStateView = ({ namespaces }: Props) => {
...
+          hideWhenSinglePage={true}
         />
-      )}
-    </>
+      }
+    >
+      {pageItems.map((rule) => {
+        const { rulerRule, promRule } = rule;
+
+        const isProvisioned = isGrafanaRulerRule(rulerRule) && Boolean(rulerRule.grafana_alert.provenance);
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/RuleListStateView.tsx:98]: not for this PR, but we could add the `isProvisionedRulerRule` function in rules.ts?

File: public/app/features/alerting/unified/rule-list/RuleList.v2.tsx [hunk 3]
```diff
@@ -4,28 +4,29 @@ import { useLocation } from 'react-router-dom-v5-compat';
 import { useAsyncFn, useInterval, useMeasure } from 'react-use';
 
 import { GrafanaTheme2, urlUtil } from '@grafana/data';
+import { logInfo } from '@grafana/runtime';
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/rule-list/RuleList.v2.tsx:7]: is this logInfo change intentional? is not correct to use the one from `'../../Analytics' `?

File: public/app/features/alerting/unified/rule-list/components/AlertRuleListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    metadata.push(
+      <MetaText icon="tag-alt">
+        <TextLink href={href} variant="bodySmall" color="primary" inline={false}>
+          {pluralize('label', labelsSize(labels), true)}
+        </TextLink>
+      </MetaText>
+    );
+  }
+
+  if (!isPaused && contactPoint) {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/rule-list/components/AlertRuleListItem.tsx:104]: maybe we could also render the contact point even if the alert rule is paused? But then we could use a different color or something to make clear that is paused and its not going to receive anything?

File: public/app/features/alerting/unified/rule-list/components/AlertRuleListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    <ListItem
+      title={
+        <Stack direction="row" alignItems="center">
+          <TextLink href={href} inline={false}>
+            {name}
+          </TextLink>
+          {origin && <PluginOriginBadge pluginId={origin.pluginId} size="sm" />}
+          {/* show provisioned badge only when it also doesn't have plugin origin */}
+          {isProvisioned && !origin && <ProvisioningBadge />}
+          {/* let's not show labels for now, but maybe users would be interested later? Or maybe show them only in the list view? */}
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/rule-list/components/AlertRuleListItem.tsx:130]: 👍

File: public/app/features/alerting/unified/rule-list/components/EvaluationGroupWithRules.tsx [hunk 6]
```diff
@@ -50,21 +50,6 @@ export const EvaluationGroupWithRules = ({ group, rulesSource }: EvaluationGroup
           );
         }
 
-        if (isRecordingRulerRule(rulerRule)) {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/rule-list/components/EvaluationGroupWithRules.tsx:53]: with this change we are not going to identify recording rule from alerting rule? or maybe i'm missing something?

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! 🚀
I added some non blocking comments 👀

File: public/app/features/alerting/unified/components/rules/RuleListStateView.tsx (hunk 2 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rules/RuleListStateView.tsx:98]: Great idea! I'll append that 👍

File: public/app/features/alerting/unified/rule-list/RuleList.v2.tsx (hunk 3 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/rule-list/RuleList.v2.tsx:7]: No this seems like a mistake, I'll fix that.

File: public/app/features/alerting/unified/rule-list/components/EvaluationGroupWithRules.tsx (hunk 6 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/rule-list/components/EvaluationGroupWithRules.tsx:53]: This component (which was used for the v2 POC) is being removed in a later PR :)

**Pull request merged**